datafusion/core/src/execution/context.rs

```diff
@@ -1401,6 +1402,9 @@ impl SessionState {
         }
         rules.push(Arc::new(ReduceOuterJoin::new()));
         rules.push(Arc::new(FilterPushDown::new()));
+        // we do type coercion after filter push down so that we don't push CAST filters to Parquet
```

Comment:
I have a partially written ticket (I will post later this week) related to supporting `CAST` in  pruning logic (which is part of what is pushed to parquet). Perhaps this is also related